digidem/comapeo-mobile
# fix: media sharing

closes #591 

### Description
- In the handleShare action for the button in an observation, only does the refetch for the query if needed (it is needed for the first time the photo is shared)
- It takes a very specific and very repeated set of actions to cause an error in the PhotoPreviewModal which I don't think anyone would ever do
- Moved the logic for photo query the PhotoPreviewModal.tsx from the PhotoUnpreparedView as we discussed @achou11 . Not sure if that is necessary or beneficial...
- I could move the photo querying logic up to the Observation/index if you think that would help. I will wait to hear your thoughts.

### To test
Create an observation
Click into it
Share it as many times as you want
You should not experience a delay
Thumbnail will not error out
